pennylane/devices/tests/test_gates.py

```diff
@@ -150,6 +205,45 @@
 # ===============================================================
 
 
+class TestSupportedGates:
+    """Test that the device can implement all gates that it supports."""
+
+    @pytest.mark.parametrize("operation", all_ops)
+    def test_supported_gates_can_be_implemented(self, device_kwargs, operation):
+        """Test that the device can implement all its supported gates."""
+        device_kwargs["wires"] = 3  # maximum size of current gates
+        dev = qml.device(**device_kwargs)
+
+        assert hasattr(dev, "operations")
+        if operation in dev.operations:
+
+            @qml.qnode(dev)
+            def circuit():
+                ops[operation]
+                return qml.expval(qml.Identity(wires=0))
+
+            assert isinstance(circuit(), (float, tensor))
```

Comment:
Does
```python
assert isinstance(circuit(), (float, np.ndarray))
```
not work? It should, since `qml.numpy.tensor` is a subclass of `ndarray` (and it will avoid the import above)

Reply:
Oh sure!